### ISSUE TITLE
Broken pusher and DB-errors after migrating from sqlite to postgresql

### ISSUE THREAD

**Issue by msirringhaus** [matrix-org/synapse#6378]
This is a spin-off from #6360 

### Description

After upgrading from 1.4 to 1.5.1 I got various problems (see #6360), so I decided to migrate from sqlite version `3.16.2` to postgresql `9.6.13`.

After manually fixing #6302, I now get these error-messages repeatedly:

```
 PUT-20979 - [TXN OPERROR] {add_messages_to_device_inbox-35b2d} could not serialize access due to concurrent update
 Nov 18 08:03:51 server python[3166]:  0/5
 PUT-20981 - [TXN OPERROR] {add_messages_to_device_inbox-35b2f} could not serialize access due to concurrent update
 Nov 18 08:03:51 server python[3166]:  0/5
 PUT-20982 - [TXN OPERROR] {add_messages_to_device_inbox-35b31} could not serialize access due to concurrent update
 Nov 18 08:03:51 server python[3166]:  0/5
 PUT-20982 - [TXN OPERROR] {add_messages_to_device_inbox-35b31} could not serialize access due to concurrent update
 Nov 18 08:03:51 server python[3166]:  1/5
 GET-21374 - Invalid JSON in data for pusher 14888: \x7b226272616[..snip..]f74227d, Expecting value: line 1 column 1 (char 0)
 GET-21407 - Invalid JSON in data for pusher 14888: \x7b226272616[..snip..]f74227d, Expecting value: line 1 column 1 (char 0)
 GET-21411 - Invalid JSON in data for pusher 14888: \x7b226272616[..snip..]f74227d, Expecting value: line 1 column 1 (char 0)
 GET-21452 - Invalid JSON in data for pusher 14888: \x7b226272616[..snip..]f74227d, Expecting value: line 1 column 1 (char 0)
 GET-21460 - Invalid JSON in data for pusher 14888: \x7b226272616[..snip..]f74227d, Expecting value: line 1 column 1 (char 0)
 GET-21474 - Invalid JSON in data for pusher 14888: \x7b226272616[..snip..]f74227d, Expecting value: line 1 column 1 (char 0)
 GET-21476 - Invalid JSON in data for pusher 14888: \x7b226272616[..snip..]f74227d, Expecting value: line 1 column 1 (char 0)
 GET-21574 - Invalid JSON in data for pusher 14888: \x7b226272616[..snip..]f74227d, Expecting value: line 1 column 1 (char 0)
 GET-21593 - Invalid JSON in data for pusher 14888: \x7b226272616[..snip..]f74227d, Expecting value: line 1 column 1 (char 0)
 GET-21599 - Invalid JSON in data for pusher 14888: \x7b226272616[..snip..]f74227d, Expecting value: line 1 column 1 (char 0)
 persist_presence_changes-7334 - [TXN OPERROR] {update_presence-38335} could not serialize access due to concurrent update
 Nov 18 13:32:50 server python[3166]:  0/5
 GET-23219 - Invalid JSON in data for pusher 14888: \x7b226272616[..snip..]f74227d, Expecting value: line 1 column 1 (char 0)
 GET-23223 - Invalid JSON in data for pusher 14888: \x7b226272616[..snip..]f74227d, Expecting value: line 1 column 1 (char 0)
 GET-23225 - Invalid JSON in data for pusher 14888: \x7b226272616[..snip..]f74227d, Expecting value: line 1 column 1 (char 0)
 GET-23229 - Invalid JSON in data for pusher 14888: \x7b226272616[..snip..]f74227d, Expecting value: line 1 column 1 (char 0)
 GET-23723 - Invalid JSON in data for pusher 14888: \x7b226272616[..snip..]f74227d, Expecting value: line 1 column 1 (char 0)
 GET-23725 - Invalid JSON in data for pusher 14888: \x7b226272616[..snip..]f74227d, Expecting value: line 1 column 1 (char 0)
 handle_presence_timeouts-90881 - [TXN OPERROR] {update_presence-39ee2} could not serialize access due to concurrent update
 Nov 18 17:30:50 server python[3166]:  0/5
```

Some push-notifications are not going through.
Also, these DB-errors worry me a bit.

### Steps to reproduce

- sqlite-database with version 1.4
- upgrade to 1.5.1
- Fix #6302 manually (adjusting the `hidden`-flag from 'FALSE' to `0`)
- Migrate to postgresql according to [this](https://github.com/matrix-org/synapse/blob/master/docs/postgres.md)

<!--
Describe how what happens differs from what you expected.

If you can identify any relevant log snippets from _homeserver.log_, please include
those (please be careful to remove any personal or private data). Please surround them with
``` (three backticks, on a line on their own), so that they are formatted legibly.
-->

### Version information

<!-- IMPORTANT: please answer the following questions, to help us narrow down the problem -->

If not matrix.org:

<!--
 What version of Synapse is running?

You can find the Synapse version with this command:

$ curl http://localhost:8008/_synapse/admin/v1/server_version

(You may need to replace `localhost:8008` if Synapse is not configured to
listen on that port.)
-->
- **Version**: 1.5.1

- **Install method**: pip
<!-- examples: package manager/git clone/pip  -->

- **Platform**: raspberry pi - bare metal (virtualenv)
<!--
Tell us about the environment in which your homeserver is operating
distro, hardware, if it's running in a vm/container, etc.
-->

**Comment by msirringhaus** [comment]
The pusher-errors seem to come from this bug #5312 

I looked into the old sqlite-DB in the table `pushers`.
Most pushers seem to have the same entry: `{"format":"event_id_only","url":"https://matrix.org/_matrix/push/v1/notify"}`.
Then I manually updated the table `pushers` in the postgresql-DB, like so:
```
update pushers set data = $${"format":"event_id_only","url":"https://matrix.org/_matrix/push/v1/notify"}$$ where id = [...snip...]
```
for each affected `id`.

Now the error-messages seem to be gone, but I still have to wait, if all push-notifications are working again.
